bench-vortex/src/engines/ddb/mod.rs

```diff
@@ -0,0 +1,187 @@
+        repo_root.unwrap_or_default()
+    ))
+    .expect("failed to create DuckDB executable path");
+
+    validate_path(&duckdb_path);
+
+    duckdb_path
+}
+
+fn create_table_registration(base_url: &Url, extension: &str, dataset: BenchmarkDataset) -> String {
+    // Base path contains trailing /.
+    let base_dir = base_url.as_str();
+    let base_dir = base_dir.strip_prefix("file://").unwrap_or(base_dir);
+
+    match dataset {
+        BenchmarkDataset::TpcH => {
+            let mut commands = String::new();
+            let tables = [
+                "customer", "lineitem", "nation", "orders", "part", "partsupp", "region",
+                "supplier",
+            ];
+
+            for table in &tables {
+                let table_path = format!("{base_dir}{table}.{extension}");
+                commands.push_str(&format!(
+                    "CREATE VIEW {table} AS SELECT * FROM read_{extension}('{table_path}');\n"
+                ));
+            }
+            commands
+        }
+        BenchmarkDataset::ClickBench { single_file } => {
+            let file_glob = if single_file {
+                format!("{base_dir}hits.{extension}")
+            } else {
+                format!("{base_dir}*.{extension}")
+            };
+
+            format!("CREATE VIEW hits AS SELECT * FROM read_{extension}('{file_glob}');")
+        }
+    }
+}
+
+/// Resolves the storage URL based on dataset and format requirements
+fn resolve_storage_url(base_url: &Url, file_format: Format, dataset: BenchmarkDataset) -> Url {
+    if file_format == Format::OnDiskVortex {
+        match dataset.vortex_path(base_url) {
+            Ok(vortex_url) => {
+                // Check if the directory exists (for file:// URLs)
+                if vortex_url.scheme() == "file" {
+                    let path = std::path::Path::new(vortex_url.path());
+                    if !path.exists() {
+                        log::warn!(
+                            "Vortex directory doesn't exist at: {}. Run with DataFusion engine first to generate Vortex files.",
+                            path.display()
+                        );
+                    }
+                }
+                vortex_url
+            }
+            Err(_) => base_url.clone(),
+        }
+    } else if file_format == Format::Parquet {
+        match dataset.parquet_path(base_url) {
+            Ok(parquet_url) => parquet_url,
+            Err(_) => base_url.clone(),
+        }
+    } else {
+        base_url.clone()
+    }
+}
+
+/// Execute DuckDB queries for benchmarks
+pub fn execute_query(
+    queries: &[String],
+    base_url: &Url,
+    file_format: Format,
+    dataset: BenchmarkDataset,
+    duckdb_path: &std::path::Path,
+) -> anyhow::Result<Duration> {
+    let extension = match file_format {
+        Format::Parquet => "parquet",
+        Format::OnDiskVortex => "vortex",
+        other => vortex_panic!("Format {other} isn't supported for DuckDB"),
+    };
```

Comment:
Should there to a `to_ext_str` method on format, and here you can do a `matches!(...)`

Reply:
This logic is duplicated many times, add a method on Format for this